frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx

```diff
@@ -273,6 +278,23 @@ const LogsTableInner = ({
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [])
 
+	useEffect(() => {
+		setTimeout(() => {
+			if (!loadingAfter && bodyRef?.current) {
+				fetchMoreWhenScrolled(bodyRef.current, true)
+			}
+		}, 0)
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [loadingAfter])
+
+	const handleFetchMoreWhenScrolled = (
```

Comment:
Couldn't use `useRef` here because it wouldn't respect state changes when a new query was fetched (i.e. `hasNextPage` was always true since it was initialized this way)